modules/uhttp/filters_test.go

```diff
@@ -29,36 +29,40 @@ import (
 	"testing"
 
 	"go.uber.org/fx"
+	"go.uber.org/fx/internal/fxcontext"
 	"go.uber.org/fx/service"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestExecutionChain(t *testing.T) {
-	chain := newExecutionChain([]Filter{}, getNoopHandler())
+func TestFilterChain(t *testing.T) {
+	host := service.NullHost()
+	chain := newFilterChain([]Filter{}, getNoopFilter(host))
 	response := testServeHTTP(chain)
 	assert.True(t, strings.Contains(response.Body.String(), "filters ok"))
 }
 
-func TestExecutionChainFilters(t *testing.T) {
-	chain := newExecutionChain(
-		[]Filter{FilterFunc(tracingServerFilter), FilterFunc(panicFilter)},
-		getNoopHandler(),
-	)
+func TestFilterChainFilters(t *testing.T) {
+	host := service.NullHost()
+	chain := newFilterChain([]Filter{
+		contextFilter(host),
+		tracingServerFilter(host),
+		panicFilter(host)},
+		getNoopFilter(host))
 	response := testServeHTTP(chain)
 	assert.Contains(t, response.Body.String(), "filters ok")
 }
 
-func testServeHTTP(chain executionChain) *httptest.ResponseRecorder {
+func testServeHTTP(chain filterChain) *httptest.ResponseRecorder {
 	request := httptest.NewRequest("", "http://filters", nil)
 	response := httptest.NewRecorder()
-	ctx := fx.NewContext(context.Background(), service.NullHost())
+	ctx := fxcontext.New(context.Background(), service.NullHost())
 	chain.ServeHTTP(ctx, response, request)
 	return response
 }
 
-func getNoopHandler() Handler {
-	return HandlerFunc(func(ctx fx.Context, w http.ResponseWriter, r *http.Request) {
+func getNoopFilter(host service.Host) HandlerFunc {
```

Comment:
Minor: I think the previous handler naming actually helped distinguish the final handler. I prefer that for readability.